pkg/collector/python/inter.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build python
+
+package python
+
+import "sync"
+
+const maxInternerStrings = 1000
+
+type stringInterner struct {
+	m map[string]string
+}
+
+func newStringInterner() *stringInterner {
+	return &stringInterner{m: make(map[string]string)}
+}
+
+func (si *stringInterner) intern(s []byte) string {
+	if interned, ok := si.m[string(s)]; ok {
+		return interned
+	}
+	ss := string(s)
+	si.m[ss] = ss
+	return ss
+}
+
+var sharedInternerMu sync.Mutex
```

Comment:
Note: I considered a `sync.Pool` here as well, but I suspect that this mutex won't see much, if any, contention due to Python's GIL and limited amount of pressure it will be able to exert on this.